caluma/core/filters.py

```diff
@@ -217,17 +218,161 @@ def filter(self, qs, value):
+        if converted:
+            return converted
 
-@convert_form_field.register(MetaValueFilterField)
-def convert_meta_value_field(field):
-    registry = get_global_registry()
-    converted = registry.get_converted_field(field)
-    if converted:
+        converted = MetaValueFilterType()
+        registry.register_converted_field(field, converted)
         return converted
 
-    converted = MetaValueFilterType()
-    registry.register_converted_field(field, converted)
-    return converted
+
+class AnswerLookupMode(Enum):
+    EXACT = "exact"
+    STARTSWITH = "startswith"
+    CONTAINS = "contains"
+    ICONTAINS = "icontains"
+
+
+class AnswerHierarchyMode(Enum):
+    DIRECT = "DIRECT"
+    FAMILY = "FAMILY"
+
+
+class HasAnswerFilterType(InputObjectType):
+    """Lookup type to search document structures
+
+    The question is either a "plain" question slug, or of the form
+    "parent_slug.question_slug". Note that in this case, the parent_slug will
+    match the form slug, which in turn may be a subform of the whole form
+    structure.
+
+    What does NOT work is matching a full path, as the lookup would quickly
+    generate very complex database queries.
+
+    So, given the document structure "top.some_form.subform.question_foo", you can
+    either search for "subform.question_foo" (if question_foo is used in other
+    contexts within the same form), or search directly for "question_foo" if
+    you don't need to be that specific, which will speed up the query slightly.
+
+    """
+
+    question = graphene.String(required=True)
+    value = graphene.types.generic.GenericScalar(required=True)
+    lookup = AnswerLookupMode()
+    hierarchy = AnswerHierarchyMode()
+
+
+class HasAnswerFilterField(forms.MultiValueField):
+    def __init__(self, label, **kwargs):
+        super().__init__(fields=(forms.CharField(), forms.CharField()))
+
+    def clean(self, data):
+        # override parent clean() which would reject our data structure.
+        # We don't validate, as the structure is already enforced by the
+        # schema.
+        return data
+
+
+class HasAnswerFilter(Filter):
+    field_class = HasAnswerFilterField
+
+    VALID_LOOKUPS = {
+        "text": [
+            AnswerLookupMode.EXACT,
+            AnswerLookupMode.STARTSWITH,
+            AnswerLookupMode.CONTAINS,
+            AnswerLookupMode.ICONTAINS,
+        ],
+        "multiple_choice": [AnswerLookupMode.EXACT, AnswerLookupMode.CONTAINS],
+    }
+
+    def filter(self, qs, value):
+        if value in EMPTY_VALUES:
+            return qs
+
+        for expr in value:
+            qs = self.apply_expr(qs, expr)
+        return qs
+
+    def apply_expr(self, qs, expr):
+        question_slug = expr["question"]
+        match_value = expr["value"]
+
+        # hierarchy lookup mode
+        hierarchy = expr.get("hierarchy", AnswerLookupMode.FAMILY)
+        if not hierarchy:
+            hierarchy = AnswerLookupMode.FAMILY
```

Comment:
Isn't this case already covered by the second argument of `get`? (same below)

Reply:
get() could return None still, that's why